File: Maths_And_Stats/Statistics/Standard_Deviation/Standard_Deviation.py
"""
author- Ose Oaiya (esoiya)

submission description: The standard deviation is a measure of the amount of variation or dispersion of a set of values
"""

import math

def variance(data, ddof=0):
    """
    Calculate the variance

    param data: data values
    type: list

    param ddof: degrees of freedom
    type: integer
    default: 0 (sample variance calculation)

    set ddof=1 to calculate population variance
    """

    # number of observations
    num = len(data)
    mean = sum(data) / num

    # square the deviations
    deviations = [(x - mean) ** 2 for x in data]

    return sum(deviations) / (num - ddof)


def standard_deviation(data, ddof=0):
    """
    Calculate the standard deviation of the data

    param data: data values
    type: list

    param ddof: degree of freedom
    type: int

    default: 0 (sample standard deviation calculation)

    set ddof=1 to calculate population standard deviation
    """

    var = variance(data, ddof)

    return math.sqrt(var)

File: Maths_And_Stats/Statistics/Standard_Deviation/Standard_Deviation.cpp
#include <iostream>
#include <cmath>
using namespace std;

float standardDeviation(float data[])
{
    float sum = 0.0, mean, standardDeviation = 0.0;

    int i;

    for(i = 0; i < 10; ++i)
    {
        sum += data[i];
    }

    mean = sum/10;

    for(i = 0; i < 10; ++i)
        standardDeviation += pow(data[i] - mean, 2);

    return sqrt(standardDeviation / 10);
};

int main()
{
    int i;
    float data[10];

    cout << "Enter 10 elements: ";
    for(i = 0; i < 10; ++i)
        cin >> data[i];

    cout << endl << "Standard Deviation = " << standardDeviation(data);

    return 0;
}

